packages/types/src/sign-client/engine.ts

```diff
@@ -390,6 +396,12 @@ export abstract class IEngine {
 
   public abstract authenticate: (
     params: AuthTypes.SessionAuthenticateParams,
+    walletUniversalLink?: string,
+  ) => Promise<EngineTypes.SessionAuthenticateResponsePromise>;
+
+  public abstract authenticateLinkMode: (
+    params: AuthTypes.SessionAuthenticateParams,
+    walletUniversalLink: string,
```

Comment:
these two methods seem the same except the name, why do we need both?

Reply:
we dont. But i'll need to move the logic that checks in localstorage for the wallet list that support link mode, that nowadays is in `sign-client/src/client/authenticate`